File: ModuleFSharp/Command1.fs
namespace ModuleFSharp
open FarNet
open System
open System.Data.Common

[<ModuleCommand(Name = "ModuleFSharp", Prefix = "ModuleFSharp", Id = "0d526cb9-7006-4ab4-8b82-7bc5d3cb8e2b")>]
type Command1() =
    inherit ModuleCommand()

    override _.Invoke(_, e) =
        // The format of command line is up to the module.
        // Here is the parser of connection string format.
        let parameters = Command1.ParseParameters(e.Command)

        Far.Api.UI.WriteLine($"Parsed {parameters.Count} parameters:")
        for key in parameters.Keys do
            Far.Api.UI.WriteLine($"{key} = {parameters[string key]}");

    static member ParseParameters(text) : DbConnectionStringBuilder =
        try
            DbConnectionStringBuilder(ConnectionString = text);
        with ex ->
            raise (ModuleException($"Use semicolon separated key=value pairs. Error: {ex.Message}"));

File: ModuleCSharp/Command1.cs
using FarNet;
using System;
using System.Data.Common;

namespace ModuleCSharp;

[ModuleCommand(Name = "ModuleCSharp", Prefix = "ModuleCSharp", Id = "30bc4a2c-6e28-43cb-ad38-5c0ee6b7f28f")]
public class Command1 : ModuleCommand
{
	public override void Invoke(object sender, ModuleCommandEventArgs e)
	{
		// The format of command line is up to the module.
		// Here is the parser of connection string format.
		var parameters = ParseParameters(e.Command);

		Far.Api.UI.WriteLine($"Parsed {parameters.Count} parameters:");
		foreach (string key in parameters.Keys)
			Far.Api.UI.WriteLine($"{key} = {parameters[key]}");
	}

	static DbConnectionStringBuilder ParseParameters(string text)
	{
		try
		{
			return new DbConnectionStringBuilder { ConnectionString = text };
		}
		catch (Exception ex)
		{
			throw new ModuleException($"Use semicolon separated key=value pairs. Error: {ex.Message}");
		}
	}
}
